### PR TITLE
Avoid DeprecationWarning from inspect.getargspec (rebased)

### PR DESCRIPTION
**Pull request by frexvahi**
Pull request #6817 broke the build because of incompatible changes which had been added to master in the intervening time. The code here has been rebased on the latest master.

### CODE REVIEW

File: docs/autogen.py [hunk 1]
```diff
@@ -320,9 +320,11 @@ def get_classes_ancestors(classes):
 
 
 def get_function_signature(function, method=True):
-    signature = getattr(function, '_legacy_support_signature', None)
-    if signature is None:
+    wrapped = getattr(function, '_legacy_wrapped_original', None)
```
Comment (fchollet) [inline docs/autogen.py]: Rename to something more explicit, such as "_original_function"

File: keras/legacy/interfaces.py [hunk 2]
```diff
@@ -86,7 +85,7 @@ def wrapper(*args, **kwargs):
                 warnings.warn('Update your `' + object_name +
                               '` call to the Keras 2 API: ' + signature, stacklevel=2)
             return func(*args, **kwargs)
-        wrapper._legacy_support_signature = inspect.getargspec(func)
+        wrapper._legacy_wrapped_original = func
```
Comment (fchollet) [inline keras/legacy/interfaces.py]: Same, please rename

**Comment by frexvahi** [comment]
The Travis build hasn't started for some reason. Could you reset it?

**Comment by fchollet** [comment]
LGTM

**Pull request merged**